src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,296 @@
+package com.minecolonies.coremod.entity.ai.citizen.builder;
+
+import com.minecolonies.coremod.blocks.AbstractBlockHut;
+import com.minecolonies.coremod.blocks.ModBlocks;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.workorders.WorkOrderBuild;
+import com.minecolonies.coremod.colony.workorders.WorkOrderBuildDecoration;
+import com.minecolonies.coremod.util.BlockUtils;
+import com.minecolonies.coremod.util.StructureWrapper;
+import net.minecraft.block.BlockHorizontal;
+import net.minecraft.block.properties.PropertyDirection;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Helper class to place and remove constructionTapes from the buildings.
+ */
+public final class ConstructionTapeHelper
+{
+    public static final  PropertyDirection FACING     = BlockHorizontal.FACING;
+
+    /**
+     * Private Constructor to hide implicit one.
+     */
+    private ConstructionTapeHelper()
+    {
+        /**
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Check if a block is placeable and return new Y position
+     * @param x Block X position
+     * @param y Block Y position
+     * @param z Block Z position
+     * @param world the world
+     * @return The new Y position
+     */
+
+    public static int checkIfPlaceable(@NotNull int x, @NotNull int y, @NotNull int z, @NotNull World world)
+    {
+        int newY = y;
+        boolean working = true;
+        while (working)
+
+        {
+            final BlockPos block = new BlockPos(x, newY, z);
+            final BlockPos blockMin1 = new BlockPos(x, newY-1, z);
+            if (world.getBlockState(block).isFullBlock() || world.getBlockState(block).getMaterial().isLiquid()) {newY = newY+1;}
+            else if (world.getBlockState(blockMin1).isFullBlock() || world.getBlockState(blockMin1).getMaterial().isLiquid()) {working = false;}
+            else {newY = newY-1;}
+        }
+        return newY;
+    }
+
+    /**
+     * Proxy to place the tape also with the building only.
+     * @param building the building.
+     * @param world the world.
+     */
+    public static void placeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+        placeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    public static void placeConstructionTape(@NotNull WorkOrderBuild workOrder, @NotNull World world)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)